com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Task.java

```diff
@@ -72,7 +73,7 @@ public TaskBuilder<T> disableCache() {
         }
 
         public TaskBuilder<T> addInput(IResource input) {
-            if (!task.inputs.contains(input)){
+            if (task.inputLookup.add(input)) {
```

Comment:
Use hash set for checking if element was added. 
it's O(1) instead of O(n) for List.contains()